### PR TITLE
Update interface paths and fix other warnings

### PR DESCRIPTION
**Pull request by vpascuzz**
Get rid of brutal warnings that flood the screen during compilation with this pull request.

`CMakeLists.txt`: add updated interface paths
`Root/BJetEfficiencyCorrector.cxx`: comment unused parameter
`Root/MuonEfficiencyCorrector.cxx`: comment unused parameter
`xAODAnaHelpers/BJetEfficiencyCorrector.h`: update interface paths
`xAODAnaHelpers/JetSelector.h`: update interface path
`xAODAnaHelpers/MuonSelector.h`: update interface path

### CODE REVIEW

**Comment by kratsg** [comment]
These changes are going to break release 20. You need to wrap this in `#ifdef USE_CMAKE` instead.

**Comment by vpascuzz** [comment]
Ah...I forgot about the back-compat. Will fix.

**Comment by vpascuzz** [comment]
OK, `PhysicsAnalysis/Interfaces/MuonAnalysisInterfaces` was `21.2.5`...

**Comment by vpascuzz** [comment]
Can we assume everyone is using >`21.2.6`? :)

**Comment by kratsg** [comment]
we're on 21.2.10 now, so we can migrate in rel21.

**Comment by vpascuzz** [comment]
OK.

**Comment by vpascuzz** [comment]
Not sure what test failed for `RootCore`, since all checks reported as passed.

**Comment by vpascuzz** [comment]
Something holding up this PR?

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,13 +5,16 @@
 # Declare the package name:
 atlas_subdir( xAODAnaHelpers )
 
-set( release_libs xAODTriggerCnvLib )
+set( release_libs xAODTriggerCnvLib FTagAnalysisInterfacesLib MuonAnalysisInterfacesLib )
```
Comment (kratsg) [inline CMakeLists.txt]: why is this added like this?

File: CMakeLists.txt [hunk 2]
```diff
@@ -5,13 +5,16 @@
...
+set( release_depends PhysicsAnalysis/Interfaces/FTagAnalysisInterfaces PhysicsAnalysis/Interfaces/MuonAnalysisInterfaces )
 if( ${AnalysisBase_VERSION} VERSION_GREATER 21.0 )
   # 21.X
-  set( release_libs xAODTriggerCnvLib )
+  set( release_libs xAODTriggerCnvLib FTagAnalysisInterfacesLib MuonAnalysisInterfacesLib )
+  set( release_depends PhysicsAnalysis/Interfaces/FTagAnalysisInterfaces PhysicsAnalysis/Interfaces/MuonAnalysisInterfaces )
 else()
   # 2.6.X
   set( release_libs xAODTriggerCnv )
+  set( release_depends PhysicsAnalysis/JetTagging/JetTagPerformanceCalibration/xAODBTaggingEfficiency PhysicsAnalysis/MuonID/MuonSelectorTools )
```
Comment (kratsg) [inline CMakeLists.txt]: we don't really support 2.6.X now.

File: Root/MuonEfficiencyCorrector.cxx [hunk 3]
```diff
@@ -493,7 +493,11 @@ EL::StatusCode MuonEfficiencyCorrector :: histFinalize ()
   return EL::StatusCode::SUCCESS;
 }
 
+#ifdef USE_CMAKE
```
Comment (kratsg) [inline Root/MuonEfficiencyCorrector.cxx]: this ifdef makes no sense here. The same change should be in both r20 and r21 if you're commenting this out...

File: xAODAnaHelpers/BJetEfficiencyCorrector.h [hunk 4]
```diff
@@ -12,8 +12,13 @@
 #include "PATInterfaces/ISystematicsTool.h"
 
 // external tools include(s):
+#ifdef USE_CMAKE
+#include "FTagAnalysisInterfaces/IBTaggingSelectionTool.h"
+#include "FTagAnalysisInterfaces/IBTaggingEfficiencyTool.h"
+#else // RootCore back-compat
```
Comment (kratsg) [inline xAODAnaHelpers/BJetEfficiencyCorrector.h]: no need to comment

File: xAODAnaHelpers/JetSelector.h [hunk 5]
```diff
@@ -27,7 +27,11 @@
 #include "AsgTools/AnaToolHandle.h"
 #include "JetJvtEfficiency/IJetJvtEfficiency.h"
 #include "JetInterface/IJetModifier.h"
+#ifdef USE_CMAKE
+#include "FTagAnalysisInterfaces/IBTaggingSelectionTool.h"
+#else // RootCore back-compat
```
Comment (kratsg) [inline xAODAnaHelpers/JetSelector.h]: no need to comment

File: xAODAnaHelpers/MuonSelector.h [hunk 6]
```diff
@@ -18,7 +18,11 @@
 // external tools include(s):
 #include "AsgTools/AnaToolHandle.h"
 #include "IsolationSelection/IIsolationSelectionTool.h"
+#ifdef USE_CMAKE
+#include "MuonAnalysisInterfaces/IMuonSelectionTool.h"
+#else // RootCore back-compat
```
Comment (kratsg) [inline xAODAnaHelpers/MuonSelector.h]: no need to comment

File: CMakeLists.txt (hunk 1 above)
Comment (vpascuzz) [reply CMakeLists.txt]: Because the latter two libraries are used in >21.0. Followed what was done before with `xAODTriggerCnvLib`.

File: CMakeLists.txt (hunk 2 above)
Comment (vpascuzz) [reply CMakeLists.txt]: Then the first comment (above) isn't needed. I just added to what was there, as you can see from the diff.

File: Root/MuonEfficiencyCorrector.cxx (hunk 3 above)
Comment (vpascuzz) [reply Root/MuonEfficiencyCorrector.cxx]: False. There is an `#ifdef` later in the function that makes use of `eventInfo` if not `USE_CMAKE`.

File: xAODAnaHelpers/BJetEfficiencyCorrector.h (hunk 4 above)
Comment (vpascuzz) [reply xAODAnaHelpers/BJetEfficiencyCorrector.h]: See above.

File: xAODAnaHelpers/JetSelector.h (hunk 5 above)
Comment (vpascuzz) [reply xAODAnaHelpers/JetSelector.h]: Followed suit (comment for 2.6 in `CMakeLists.txt`).

File: xAODAnaHelpers/MuonSelector.h (hunk 6 above)
Comment (vpascuzz) [reply xAODAnaHelpers/MuonSelector.h]: See above.

File: CMakeLists.txt (hunk 1 above)
Comment (kratsg) [reply CMakeLists.txt]: No, but we don't really need it. You're also defining it twice which doesn't make sense.

File: xAODAnaHelpers/BJetEfficiencyCorrector.h (hunk 4 above)
Comment (kratsg) [reply xAODAnaHelpers/BJetEfficiencyCorrector.h]: The `// RootCore back-compat` comment needs to be removed. We are planning to remove these lines later, and anything after the `#else` will screw up the removal.

File: xAODAnaHelpers/JetSelector.h (hunk 5 above)
Comment (kratsg) [reply xAODAnaHelpers/JetSelector.h]: The `// RootCore back-compat` comment needs to be removed. We are planning to remove these lines later, and anything after the `#else` will screw up the removal.

File: xAODAnaHelpers/MuonSelector.h (hunk 6 above)
Comment (kratsg) [reply xAODAnaHelpers/MuonSelector.h]: The `// RootCore back-compat` comment needs to be removed. We are planning to remove these lines later, and anything after the `#else` will screw up the removal.

File: CMakeLists.txt (hunk 1 above)
Comment (vpascuzz) [reply CMakeLists.txt]: (I wonder how the original duplicate `set` ended up merged then...)
Again, I was following what was already done so as to not change the workflow. Also not sure what you mean by "No, but we really don't need it."

File: xAODAnaHelpers/JetSelector.h (hunk 5 above)
Comment (vpascuzz) [reply xAODAnaHelpers/JetSelector.h]: Fine.

File: xAODAnaHelpers/MuonSelector.h (hunk 6 above)
Comment (vpascuzz) [reply xAODAnaHelpers/MuonSelector.h]: Fine.

File: xAODAnaHelpers/BJetEfficiencyCorrector.h (hunk 4 above)
Comment (vpascuzz) [reply xAODAnaHelpers/BJetEfficiencyCorrector.h]: Fine.

File: CMakeLists.txt (hunk 1 above)
Comment (kratsg) [reply CMakeLists.txt]: Yes, but the `set( release_libs )` is only for the differences between 2.6 and rel21. The ones your adding could've just been put into the list of dependencies rather than creating a new variable in CMake which needs to be unset.

Comment (vpascuzz) [reply CMakeLists.txt]: The `set`s are for backward-compatibility, as you had originally requested. So I don't know what you expect. The duplicated `set`s, I agree, can go.

Comment (kratsg) [reply CMakeLists.txt]: Backwards compatibility is for release 20 only.

Comment (vpascuzz) [reply CMakeLists.txt]: Sorry, I'm confused. Why is this done for `xAODTriggerCnvLib` then?
The `set( release_libs )` is for differences between 2.6 and rel21, yes. There is no `FTagAnalysisInterfacesLib` or `MuonAnalysisInterfacesLib` in 2.6, is there?

Comment (vpascuzz) [reply CMakeLists.txt]: `MuonAnalysisInterfacesLib` and `FTagAnalysisInterfacesLib` and were introduced in 21.2.5 and 21.2.6, respectively.

Comment (kratsg) [reply CMakeLists.txt]: > Sorry, I'm confused. Why is this done for xAODTriggerCnvLib then?
> The set( release_libs ) is for differences between 2.6 and rel21, yes. There is no FTagAnalysisInterfacesLib or MuonAnalysisInterfacesLib in 2.6, is there?

2.6.X isn't supported at all. Rel20 and Rel21 are the only things that are supported. The flag was left there from when we supported it before.

Comment (vpascuzz) [reply CMakeLists.txt]: So you suggest adding the two libraries to the depends list, even though they aren't present in rel20?

Comment (kratsg) [reply CMakeLists.txt]: > So you suggest adding the two libraries to the depends list, even though they aren't present in rel20?

CMakeLists.txt is only used in release 21. https://github.com/UCATLAS/xAODAnaHelpers/blob/master/cmt/Makefile.RootCore is used in release 20.

Comment (vpascuzz) [reply CMakeLists.txt]: Right. Thanks for the reminder...

**Comment by kratsg** [comment]
@vpascuzz lemme know when you get these changes in.

**Comment by vpascuzz** [comment]
Hopefully this settles it.

Thanks for the discussion, @kratsg.

**Comment by vpascuzz** [comment]
#highfive
